GHCN.py

```diff
@@ -0,0 +1,1044 @@
+
+## load pre-processed EC station climatology
+def loadGHCN(): 
+  ''' Load a pre-processed GHCN station climatology. '''
+  return NotImplementedError
+loadGHCN_Stn = loadGHCN
+
+
+## some helper functions to test conditions
+# defined in module main to facilitate pickling
+#def test_prov(val,index,dataset,axis):
+#  ''' check if station province is in provided list ''' 
+#  return dataset.stn_prov[index] in val
+def test_begin(val,index,dataset,axis):
+  ''' check if station record begins before given year ''' 
+  return dataset.stn_begin_date[index] <= val # converted to month beforehand 
+def test_end(val,index,dataset,axis):
+  ''' check if station record ends after given year ''' 
+  return dataset.stn_end_date[index] >= val # converted to month beforehand 
+def test_minlen(val,index,dataset,axis):
+  ''' check if station record is longer than a minimum period ''' 
+  return dataset.stn_rec_len[index] >= val 
+def test_maxzse(val,index,dataset,axis, lcheckVar=True):
+  ''' check that station elevation error does not exceed a threshold ''' 
+  if not dataset.hasVariable('zs_err'):
+    if lcheckVar: raise DatasetError
+    else: return True # EC datasets don't have this field...
+  else: return np.abs(dataset.zs_err[index]) <= val
+def test_maxz(val,index,dataset,axis, lcheckVar=True):
+  ''' check that station elevation does not exceed a threshold ''' 
+  if not dataset.hasVariable('stn_zs'):
+    if lcheckVar: raise DatasetError
+    else: return True # EC datasets don't have this field...
+  else: return np.abs(dataset.stn_zs[index]) <= val
+def test_lat(val,index,dataset,axis):
+  ''' check if station is located within selected latitude band '''
+  return val[0] <= dataset.stn_lat[index] <= val[1] 
+def test_lon(val,index,dataset,axis):
+  ''' check if station is located within selected longitude band ''' 
+  return val[0] <= dataset.stn_lon[index] <= val[1] 
+def test_cluster(val,index,dataset,axis, cluster_name='cluster_id', lcheckVar=True):
+  ''' check if station is located within selected longitude band '''
+  if not dataset.hasVariable(cluster_name):
+    if lcheckVar: raise DatasetError
+    else: return True # most datasets don't have this field...
+  elif isinstance(val, (int,np.integer)): 
+    return dataset[cluster_name][index] == val
+  elif isinstance(val, (tuple,list,np.ndarray)):
+    return dataset[cluster_name][index] in val
+  else: ValueError, val
+# apply tests to list
+def apply_test_suite(tests, index, dataset, axis):
+  ''' apply an entire test suite to '''
+  # just call all individual tests for given index
+  results = [test(index,dataset,axis) for test in tests]
+  results = [np.all(res) if isinstance(res,np.ndarray) else res for res in results]  
+  return all(results)
+
+## select a set of common stations for an ensemble, based on certain conditions
+def selectStations(datasets, stnaxis='station', master=None, linplace=False, lall=False, 

```

Comment:
Did you change anything related to station selection? I'm considering to move this whole section into a separate module that can be imported from both, the EC and GHCN module. This will make it easier to share tests for station selection.